recipes/avahi/all/conanfile.py

```diff
@@ -26,75 +30,68 @@ class AvahiConan(ConanFile):
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    @property
-    def _configure_args(self):
-        yes_no = lambda v: "yes" if v else "no"
-        return [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-            "--disable-gtk3",
-            "--disable-mono",
-            "--disable-python",
-            "--disable-qt5",
-            "--disable-monodoc",
-            "--enable-compat-libdns_sd",
-            "--with-systemdsystemunitdir={}/lib/systemd/system".format(self.package_folder),
-        ]
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.configure(configure_dir=self._source_subfolder, args=self._configure_args)
-        return self._autotools
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--enable-compat-libdns_sd")
+        tc.configure_args.append("--disable-gtk3")
+        tc.configure_args.append("--disable-mono")
+        tc.configure_args.append("--disable-monodoc")
+        tc.configure_args.append("--disable-python")
+        tc.configure_args.append("--disable-qt5")
+        tc.configure_args.append("--with-systemdsystemunitdir=/lib/systemd/system")
+        tc.generate()
 
     def build(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.install()
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        tools.rmdir(os.path.join(self.package_folder, "etc"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        rmdir(self, os.path.join(self.package_folder, "etc"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "run"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "Avahi"
         self.cpp_info.names["cmake_find_package_multi"] = "Avahi"
```

Comment:
I think it could be removed. There is reference for avahi on its project. Only Hunter uses Avahi, but still, it's not official.

Reply:
Done.